fog/view/server/src/block_tracker.rs

```diff
@@ -568,233 +467,221 @@ mod tests {
+                block_tracker.highest_fully_processed_block_count(&[rec1.clone(), rec2.clone()]),
+                (min(rec2.status.start_block, i + 1), None)
+            );
+        }
+
+        // At this point, the reason we are blocked is rec2
+        for i in 10..20 {
+            block_tracker.block_processed(rec1.key, i);
+
+            assert_eq!(
+                block_tracker.highest_fully_processed_block_count(&[rec1.clone(), rec2.clone()]),
+                (min(rec2.status.start_block, i + 1), Some(rec2.clone()))
             );
         }
 
         // Advancing the second range would get us all the way to the first one and stop there.
-        for i in 0..40 {
-            block_tracker.block_processed(ingestable_range2.id, ingestable_range2.start_block + i);
+        for i in 0..10 {
+            block_tracker.block_processed(rec2.key, rec2.status.start_block + i);
+
+            // Note: We advanced rec1 20 times in previous loop
+            // The reason we are blocked changes once we reach block 20, because then neither record is slower
+            let expected = if i == 9 {
+                (20, None)
+            } else {
+                (rec2.status.start_block + i + 1, Some(rec2.clone()))
+            };
+            assert_eq!(
+                block_tracker.highest_fully_processed_block_count(&[rec1.clone(), rec2.clone()]),
+                expected,
+                "i = {}",
+                i
+            );
+        }
 
+        // After this point, rec1 is the slow one, so that is the reason we are blocked
+        for i in 10..30 {
+            block_tracker.block_processed(rec2.key, rec2.status.start_block + i);
+
+            let expected = (20, Some(rec1.clone()));
             assert_eq!(
-                block_tracker.highest_fully_processed_block_count(
-                    &[ingestable_range1.clone(), ingestable_range2.clone()],
-                    &[]
-                ),
-                min(
-                    ingestable_range1.start_block + 20, // We advanced the first range 20 times in the previous loop
-                    ingestable_range2.start_block + i + 1
-                ),
+                block_tracker.highest_fully_processed_block_count(&[rec1.clone(), rec2.clone()]),
+                expected,
+                "i = {}",
+                i
             );
         }
+
+        // Test that marking rec1 lost enables us to progress all the way to rec2
+        rec1.status.lost = true;
+        let expected = (40, None);
+        assert_eq!(
+            block_tracker.highest_fully_processed_block_count(&[rec1.clone(), rec2.clone()]),
+            expected,
+        );
+    }
+
+    // A block tracker with a multiple ingestable ranges waits for both of them,
+    // but makes progress when the slow one is reported lost
+    #[test_with_logger]
+    #[ignore]
+    fn highest_fully_processed_block_tracks_multiple_recs_some_lost1(_logger: Logger) {
```

Comment:
Good test case, commenting to increase the chances of not forgetting to implement it :)